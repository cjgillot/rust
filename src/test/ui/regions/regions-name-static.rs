struct Foo<'static> {
    //~^ ERROR invalid lifetime parameter name: `'static`
    //~| ERROR parameter `'static` is never used [E0392]
    x: &'static isize,
}

fn main() {}
