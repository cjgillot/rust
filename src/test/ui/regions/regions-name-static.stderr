error[E0262]: invalid lifetime parameter name: `'static`
  --> $DIR/regions-name-static.rs:1:12
   |
LL | struct Foo<'static> {
   |            ^^^^^^^ 'static is a reserved lifetime name

error[E0392]: parameter `'static` is never used
  --> $DIR/regions-name-static.rs:1:12
   |
LL | struct Foo<'static> {
   |            ^^^^^^^ unused parameter
   |
   = help: consider removing `'static`, referring to it in a field, or using a marker such as `PhantomData`

error: aborting due to 2 previous errors

Some errors have detailed explanations: E0262, E0392.
For more information about an error, try `rustc --explain E0262`.
